### ISSUE TITLE
Autocomplete on comments

### ISSUE THREAD

**Issue by gencer** [castwide/solargraph#247]
Is it possible to enable Solargraph module/class completions in yard docs?

For example:

```ruby
# @type [Message::Cl...] #... autocomplete here.
tx = mime_type('png)
```

Basically if we declare a type, return or similar helpers we should see completion list.

**Comment by castwide** [comment]
It should be possible. Solargraph can already detect when you're inside a comment, but its current behavior during autocompletion is to ignore them. We might be able to trigger autocompletion in tags with a simple regular expression. I'll look into it.

**Comment by castwide** [comment]
The first implementation of autocompletion on comments is in the `master` branch.

**Comment by gencer** [comment]
@castwide Works perfectly on my end.

In the future can **CTRL + Hover&Click -> Go to definition** feature be implemented? (Just a suggestion)

**Comment by castwide** [comment]
> In the future can **CTRL + Hover&Click -> Go to definition** feature be implemented? (Just a suggestion)

It should already work, at least in the simplest case of a tag with a single type:

```ruby
class Foo; end

class Bar
  # @return [Foo] <- Go to definition should work inside the tag
  def get_foo; end
end
```

...although I did notice a small bug where the above doesn't work with the cursor between the last `o` and the `]`. I'll add it to the roadmap.

**Comment by gencer** [comment]
Ah! Yes it indeed works. However what makes me think of not was that, there is no underline and hover effect on when I hover my mouse there along with CTRL. Let me show you:

![fred](https://user-images.githubusercontent.com/1711576/86934555-82865500-c144-11ea-8a94-61fddd225c10.gif)

When we hover with CTRL on B module or A class, can we highlight them? Underscore and colorization perhaps.

**Comment by castwide** [comment]
I think that's a problem with nested namespaces, e.g., `[B]` should work, but `[B::A]` does not. I'll work on that next.

**Comment by gencer** [comment]
Yes, It is indeed works with only `B` but not with nested.

**Comment by castwide** [comment]
This feature (including a fix for nested namespaces) is released in v0.39.12.
